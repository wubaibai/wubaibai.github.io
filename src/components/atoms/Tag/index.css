.tag {
	display: inline-flex;
	position: relative;
	transition: all 0.2s;

	&:hover {
		box-shadow: white 4px 4px 0 0;

		& .base {
			top: -2px;
			left: -2px;
		}
	}
}

.text {
	transition: all 0.2s;
	padding: 6px 10px;
	color: white;
	font-family: var(--fontNarrow);
	position: relative;
	z-index: 1;
	cursor: default;
}

.base {
	background: white;
	opacity: 0.2;
	position: absolute;
	top: 0;
	left: 0;
	width: 100%;
	height: 100%;
	transition: all 0.1s;
	z-index: 0;
}
