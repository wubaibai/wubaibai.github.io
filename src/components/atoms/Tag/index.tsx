import React from 'react';

import style from './index.css';

const Tag: React.FC = ({ children }) => (
	<div className={style.tag}>
		<div className={style.text}>{children}</div>
		<div className={style.base} />
	</div>
);

export default Tag;
