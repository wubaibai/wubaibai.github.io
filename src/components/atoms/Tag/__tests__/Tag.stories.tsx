import React from 'react';
import { storiesOf } from '@storybook/react';

import Tag from 'components/atoms/Tag';

const stories = storiesOf('atoms/Tag', module);

stories.add('__interactive', () => <Tag>Node.js</Tag>, { jest: 'Tag' });
