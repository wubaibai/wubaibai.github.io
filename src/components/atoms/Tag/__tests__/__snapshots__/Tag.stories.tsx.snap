// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="tag"
  >
    <div
      class="text"
    >
      Node.js
    </div>
    <div
      class="base"
    />
  </div>
</DocumentFragment>
`;
